.github/actions/flatpak-manifest-validator/action.yaml

```diff
@@ -0,0 +1,38 @@
+name: Flatpak Manifest Validator
+description: Checks order of Flatpak modules in manifest file
+inputs:
+  manifestFile:
+    description: Flatpak manifest file to check
+  failCondition:
+    description: Controls whether failed checks also fail the workflow run
+    required: false
+    default: never
+  workingDirectory:
+    description: Working directory for checks
+    required: false
+    default: ${{ github.workspace }}
+runs:
+  using: composite
+  steps:
+    - name: Check Runner Operating System 🏃‍♂️
+      if: runner.os == 'Windows'
+      shell: bash
+      run: |
+        : Check Runner Operating System 🏃‍♂️
+        echo "services-validation action requires a macOS-based or Linux-based runner."
+        exit 2
+
+    - name: Validate Flatpak Manifest 🕵️
+      shell: bash
+      working-directory: ${{ inputs.workingDirectory }}
+      run: |
+        : Validate Flatpak Manifest 🕵️
+
+        echo ::group::Run Validation
+        if [[ '${{ inputs.failCondition }}' == 'never' ]]; then set +e; fi
+        python3 -u \
+          build-aux/format-manifest.py \
+          build-aux/com.obsproject.Studio.json \
+          --check \
+          --loglevel INFO
+        echo ::endgroup::
```

Comment:
I may be mistaken, but I don't see a check if the JSON got it's format fixed (e.g. wrong tab/space).

Reply:
That's what passing the `--check` switch is for - without it, the Python script would update the manifest.

But with it, the Python script will return non-zero if any changes would be necessary.

Reply:
It mimics the other formatting checker, which actually _updates_ the formatting of the files, _except_ if `--check` is passed - the parameter servers specifically the purpose used on CI.

Reply:
> _except_ if `--check` is passed

So there is no check if spacing (4 space per indentation) is correct, which is bad.

Reply:
Ah thanks for pointing that out - updated the script accordingly. It will also throw an error if the formatting changes by having it compare json string generated with a 4-space indent.